src/main/java/com/internet/shop/controller/user/GetUserDetailsController.java

```diff
@@ -10,16 +10,17 @@
 import javax.servlet.http.HttpServletResponse;
 
 public class GetUserDetailsController extends HttpServlet {
+    private static final String USER_ID = "user_id";
     private static final Injector injector = Injector.getInstance("com.internet.shop");
     private final UserService userService =
             (UserService) injector.getInstance(UserService.class);
 
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse resp)
             throws ServletException, IOException {
-        long id = Long.parseLong(req.getParameter("id"));
+        Long userId = (Long) req.getSession().getAttribute(USER_ID);
         try {
-            req.setAttribute("user", userService.get(id));
+            req.setAttribute("user", userService.get(userId));
         } catch (NoSuchElementException exception) {
             req.getRequestDispatcher("/WEB-INF/views/error.jsp").forward(req, resp);
```

Comment:
if this method somehow throes error you can solve it without try-catch and Request dispatcher.
`<error-page>
        <location>[link on error jsp file]</location>
    </error-page>`
Insert this code into web.xml and this page will be open automatically